src/word_logits_processor.py

```diff
@@ -36,74 +32,78 @@ def __init__(self, tokenizer, num_beams, word_validator: WordValidator):
         self.tokenizer = tokenizer
         self.word_validator = word_validator
         self.num_beams = num_beams
-        self.excluded_beams_by_input_idx = defaultdict(list)
+        self.excluded_beams_by_input_idx = defaultdict(lambda: list())
         self.words_to_check_by_input_idx = defaultdict(lambda: 0)
         self.failed_sequences = set()
 
     def is_valid_beam(
         self,
-        input_idx, # input idx being processed
+        input_idx,  # input idx being processed
         sequence,  # sequence generated so far
         token_id,  # next token to be generated (argmax of beam_scores)
-        beam_scores # probability of all tokens to be generated
+        beam_scores,  # probability of all tokens to be generated
     ):
         """
-            Check whether beam is valid according to the passed validators.
+        Check whether beam is valid according to the passed validators.
 
-            To enable validating on a word-level, this method backtracks
-            to collect the predicted word when it detects that the predicted
-            subword (token) is a word ending.
+        To enable validating on a word-level, this method backtracks
+        to collect the predicted word when it detects that the predicted
+        subword (token) is a word ending.
         """
-        # Token-level checks
-        # if token_id in self.banned_token_ids:
-        #     return False
-
-        # Word-level checks
-        current_subword = self.tokenizer.decode(token_id)
-        backtrack_word = ""
-        is_subword_ending = False
-        for char in current_subword:
+
+        # Begin by checking whether tokens
+        # to-be-generated indicate a phrase ending
+        if "pegasus" in self.tokenizer.name_or_path:
```

Comment:
Ended up doing special handling for pegasus - otherwise we do unnecessary checks for BART.